### PR TITLE
limit list size so that updates run in real time

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@chiahaoliu ready for review.

**Review by mrakitin — APPROVED** [review]
Looks good to me 👍

File: xpdview/waterfall.py [hunk 1]
```diff
@@ -39,7 +39,9 @@ class Waterfall:
     def __init__(self, fig=None, canvas=None,
                  key_list=None, int_data_list=None,
                  *, unit=None, label_size=16, tick_size=14,
+                 size=100,
```
Comment (sbillinge) [inline xpdview/waterfall.py:45]: I feel we are hiding some pesky magic in here. I suggest to update the docstring to reflect this change

Comment (CJ-Wright) [reply xpdview/waterfall.py:45]: Sure. The main reason for this is that asking matplotlib to recreate a plot of 1000+ lines every time a new one comes in seems to slow it down a bunch.

Comment (mrakitin) [reply xpdview/waterfall.py:45]: Maybe @tacaswell may suggest a better approach here?

Comment (CJ-Wright) [reply xpdview/waterfall.py:45]: the docstring has been updated

**Comment by CJ-Wright** [comment]
Once this passes, I'm going to merge and tag, as we need this at the beamline ASAP. I would love to have a better solution though, since this is a bit clunky.

**Pull request merged**